crates/sui-framework/packages/dwallet-system/sources/dwallet_2pc_mpc_ecdsa_k1.move

```diff
@@ -186,6 +190,26 @@ module dwallet_system::dwallet_2pc_mpc_ecdsa_k1 {
         transfer::transfer(output, session.sender);
     }
 
+    #[allow(unused_function)]
+    /// the same as create_dkg_output, but is public for testing.
+    public fun create_dkg_output_for_testing(
```

Comment:
remove this, this is a very bad pattern. Never create a copy function just for testing, this is ALWAYS a bad idea.
You can't make sure that someone maintains both function.
When writing tests you should never modify the tested code, unless there is some need to make it more testable (without adding test-only functionality, or some side effects that can be triggered optionally - their effect does not drastically affect the performance).
Can you make `create_dkg_output pub(friend)` ?

Reply:
I agree with you, I used it only because this is the pattern introduced in here:
[Utilities with #[test_only]](https://move-book.com/move-basics/testing.html#utilities-with-test_only)

I should be able to make the function `public(friend)` and test it that way.

Reply:
My implementation of the pattern was wrong. 
Fixed it.